src/remote.c

```diff
@@ -108,6 +109,15 @@ static int write_add_refspec(git_repository *repo, const char *name, const char
 	if ((error = ensure_remote_name_is_valid(name)) < 0)
 		return error;
 
+	if ((error = git_refspec__parse(&spec, refspec, fetch)) < 0) {
+		if (giterr_last()->klass != GITERR_NOMEMORY)
```

Comment:
I was thinking the same thing when I wrote this bit. We could make this just do validation if the out param is `NULL`.